modules/tensor_mechanics/src/materials/EigenstrainHelper.C

```diff
@@ -0,0 +1,163 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "EigenstrainHelper.h"
+
+#include "Assembly.h"
+#include "MooseMesh.h"
+#include "SubProblem.h"
+
+// libmesh includes
+#include "libmesh/quadrature.h"
+
+EigenstrainHelper::EigenstrainHelper(const MooseObject * moose_object, bool bnd)
+  : _eh_params(moose_object->parameters()),
+    _eh_subproblem(*_eh_params.get<SubProblem *>("_subproblem")),
+    _eh_tid(_eh_params.get<THREAD_ID>("_tid")),
+    _eh_assembly(_eh_subproblem.assembly(_eh_tid)),
+    _eh_qrule(bnd ? _eh_assembly.qRuleFace() : _eh_assembly.qRule()),
+    _eh_JxW(bnd ? _eh_assembly.JxWFace() : _eh_assembly.JxW()),
+    _eh_coord(_eh_assembly.coordTransformation()),
+    _eh_q_point(bnd ? _eh_assembly.qPointsFace() : _eh_assembly.qPoints()),
+    _eh_ncols(1 + _eh_subproblem.mesh().dimension()),
+    _eh_second_order(_eh_subproblem.mesh().hasSecondOrderElements()),
+    _eh_eigsum(),
+    _eh_A(),
+    _eh_b(6),
+    _eh_AT(),
+    _eh_ATb(_eh_ncols),
+    _eh_x(6, DenseVector<Real>(_eh_ncols)),
+    _eh_vals(6),
+    _eh_adjusted_eigenstrain()
```

Comment:
The abbreviation `eh` could be ambiguous, eh? Perhaps the written out `_eigenstrain_helper` or `_eigenstrain_adjustement`

Reply:
It is de facto standard in similar moose interface classes to prefix with an abbreviation like this!